parsel/csstranslator.py

```diff
@@ -38,46 +48,73 @@ def __str__(self):
 
         return path
 
-    def join(self, combiner, other, *args, **kwargs):
+    def join(
+        self: _XPathExprType,
+        combiner: str,
+        other: OriginalXPathExpr,
+        *args: Any,
+        **kwargs: Any,
+    ) -> _XPathExprType:
+        assert isinstance(other, XPathExpr)
         super().join(combiner, other, *args, **kwargs)
         self.textnode = other.textnode
         self.attribute = other.attribute
         return self
 
 
+if TYPE_CHECKING:
+    # requires Python 3.8
+    from typing import Protocol
+
+    # e.g. cssselect.GenericTranslator, cssselect.HTMLTranslator
+    class TranslatorProtocol(Protocol):
+        def xpath_element(self, selector: Element) -> OriginalXPathExpr:
+            ...
```

Comment:
I like @kmike’s suggestion elsewhere to use `...` where it is a placeholder for actual code, but continue to use `pass` otherwise.